cypress/support/getFormValue.js

```diff
@@ -0,0 +1,11 @@
+Cypress.Commands.add('getFormValue', fieldName => {
+    // make sure form spy function ran
+    cy.get('.form-spy-internal')
+
+    return cy.window().then(win => {
+        // unlike in vanilla js, instead of returning
+        // the next value, in cypress we have to wrap
+        // the value
+        cy.wrap(win.formValues[fieldName])
```

Comment:
The way I read it, cypress doesn't return a promise but another object with a custom then. It should be possible though to return anything that's `thenable`: https://docs.cypress.io/api/commands/then.html#Yields

So a Promise.resolve should also work here right? Maybe that'd be a bit clearer?

Reply:
Ah, looking at the cypress docs it seems to me like wrap is also meant for this: https://docs.cypress.io/api/commands/wrap.html. In that case this seems fine 👍